modules/mapbox/src/mapbox-layer.js

```diff
@@ -9,13 +10,16 @@ export default class DeckLayer {
     this.id = props.id;
     this.type = 'custom';
     this.renderingMode = '3d';
+    this.map = null;
     this.deck = null;
     this.props = props;
   }
 
+  /* Mapbox custom layer methods */
+
   onAdd(map, gl) {
     this.map = map;
-    this.deck = getDeckInstance({map, gl});
+    this.deck = getDeckInstance({map, gl, deck: this.props.deck});
```

Comment:
Storing `_deck` on the map could avoid having to pass deck around like this...

Reply:
This is for the use case where we add a `MapboxLayer` referencing a layer from an existing `Deck` instance:

```js
new MapboxLayer({id: 'my-layer', deck});
```